homeassistant/util/logging.py

```diff
@@ -108,15 +108,15 @@ def log_exception(format_err: Callable[..., Any], *args: Any) -> None:
 
 def catch_log_exception(
     func: Callable[..., Any], format_err: Callable[..., Any], *args: Any
-) -> Callable[[], None]:
+) -> Union[Callable[..., None], Callable[..., Coroutine[Any, Any, None]]]:
```

Comment:
Have we checked that this typing is compatible with a typed coroutine function that uses this as a decorator? There's no example in our current code, but this should be able to be used as a decorator.

Reply:
I haven't.

Reply:
I've done some testing with a simplified version of this method. I've uploaded the code (and results) here: [gist.gihtub.com/cdce8p](https://gist.github.com/cdce8p/76adb5273db3d92e878af7cbe78a1b06).

I'm by no means an expert on type annotations, but what I found was that the current change probably isn't enough. To make everything checkout I needed to add `overloads` and a `cast`. Additionally I changed `Coroutine[Any, Any, None]` to `Awaitable[None]`.

To move this PR forward I'll push my changes to this branch directly. This should simplify the review.

Reply:
Thanks! Kind of pity that there's roughly the same amount of type hinting as there is actual code for this function, but :shrug: